### PR TITLE
Feature/add etcd support

### PR DESCRIPTION
**Pull request by DeathBorn**
- Default topology changed to etcd
- Topology configuration moved to global attributes
- Added etcd tests

@ernestas-vinted @ernestas-poskus @mvaitiekunas @DeathBorn

### CODE REVIEW

File: attributes/mysqlctld.rb [hunk 1]
```diff
@@ -324,10 +324,10 @@
 default['vitess']['mysqlctld']['topo_global_root'] = '/vitess/global'
 
 # the address of the global topology server
-default['vitess']['mysqlctld']['topo_global_server_address'] = 'localhost:2181'
+default['vitess']['mysqlctld']['topo_global_server_address'] = 'localhost:2379'
```
Comment (ernestas-vinted) [inline attributes/mysqlctld.rb]: you could have attribute default['vitess']['topo_global_server_address'] = 'SOME' and a helper for this global attribute and then use for each Vitess binary.

File: attributes/mysqlctld.rb [hunk 2]
```diff
@@ -324,10 +324,10 @@
...
 
 # the topology implementation to use
-default['vitess']['mysqlctld']['topo_implementation'] = 'zk2'
+default['vitess']['mysqlctld']['topo_implementation'] = 'etcd2'
```
Comment (ernestas-vinted) [inline attributes/mysqlctld.rb]: you could have attribute default['vitess']['topo_implementation'] = 'SOME' and a helper for this global attribute and then use for each Vitess binary.

File: test/integration/cookbooks/test-vitess/recipes/etcd.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
+case node['platform']
+when 'debian', 'ubuntu'
+  package %w[curl software-properties-common]
+end
+
+
+etcd_ver = 'v3.3.18'
+download_url = 'https://github.com/etcd-io/etcd/releases/download'
```
Comment (ernestas-vinted) [inline test/integration/cookbooks/test-vitess/recipes/etcd.rb]: maybe there is cookbook or RPM package for etcd?

Comment (DeathBorn) [reply test/integration/cookbooks/test-vitess/recipes/etcd.rb]: did not want to make a dependency here in a same way as with zookeeper.

Comment (DeathBorn) [reply test/integration/cookbooks/test-vitess/recipes/etcd.rb]: seems rpm versions is not too old.

File: attributes/mysqlctld.rb (hunk 1 above)
Comment (DeathBorn) [reply attributes/mysqlctld.rb]: moved to global attributes

**Pull request merged**